### DISCUSSION TITLE
logging HeifFile.data shows input file type bytes rather than heic converted bytes

### DISCUSSION THREAD

**Discussion by Barnards** [bigcat88/pillow_heif#217] [Q&A] [answered]
`heif_file = pillow_heif.from_bytes(mode='RGBA', size=(595,842), data=jpeg_bytes)`
`print(heif_file.data)`
    
   log seems to show jpeg bytes rather than heic bytes, have I missed something?

**Comment by bigcat88** [comment, marked as answer]
Can you please elaborate a bit?
*content `heif_file.data` will be equal to `jpeg_bytes`, as you code just created `HeifFile` class from provided raw image data(jpeg_bytes)*

If you want to get encoded `HEIC` or `AVIF` file from `jpeg_bytes` you need to call `heif_file.save("output.heic")`

Or you can directly call `pillow_heif.encode(mode='RGBA', size=(595,842), data=jpeg_bytes, "output.heic")`

**Reply by Barnards** [reply]
Thanks for the quick response, I'm new to all things image code related. I had initially gotten a bug when I had attempted `.encode()` before I approached you, turns out there was a mismatch in jpeg size and format vs the size and format I was attempting to encode. With your advice on going the encode route I've had success with this code:

`def convert_jpeg_bytes_to_heic_bytes(jpeg_bytes):

    heic_file_name = 'file_conversion_test.heic'
    input_stream = BytesIO(jpeg_bytes)
    img = Image.open(input_stream)

    if img.mode != 'RGBA':
        img = img.convert('RGBA')

    raw_data = img.tobytes()

    pillow_heif.encode(
        mode=img.mode, 
        size=img.size, 
        data=raw_data,
        fp=heic_file_name
    )

    pillow_heif.register_heif_opener()

    with open(heic_file_name, "rb") as file:
        heic_bytes = file.read()

    input_stream.close()

    os.remove(heic_file_name)

    return heic_bytes`
    
Grateful for this library
